File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int cnt = 0;
        long ans = 0;
        int n = nums.length;
        int i = 0;
        int max = Integer.MIN_VALUE;

        for (var x : nums)
            if (x > max)
                max = x;

        for (int j = 0; j < n; j++) {
            if (nums[j] == max) {
                cnt++;
            }
            while (cnt >= k) {
                if (nums[i++] == max) {
                    cnt--;
                }
                ans += (n - j);
            }
        }
        return ans;
    }
}

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        maxV = max(nums)
        i = ans = temp = 0
        n = len(nums)

        for x in nums:
            if x >= maxV:
                temp += 1
            while temp >= k:
                if nums[i] == maxV:
                    temp -= 1
                i += 1
            ans += i
        return ans
